wiring/src/spark_wiring_fuel.cpp

```diff
@@ -175,23 +189,23 @@ void FuelGauge::readConfigRegister(byte &MSB, byte &LSB) {
 
 
 void FuelGauge::readRegister(byte startAddress, byte &MSB, byte &LSB) {
-#if Wiring_Wire3
-	Wire3.beginTransmission(MAX17043_ADDRESS);
-    Wire3.write(startAddress);
-    Wire3.endTransmission(true);
-
-    Wire3.requestFrom(MAX17043_ADDRESS, 2, true);
-    MSB = Wire3.read();
-    LSB = Wire3.read();
-#endif
+  if (_i2c) {
+    _i2c->beginTransmission(MAX17043_ADDRESS);
+    _i2c->write(startAddress);
+    _i2c->endTransmission(true);
+
+    _i2c->requestFrom(MAX17043_ADDRESS, 2, true);
+    MSB = _i2c->read();
+    LSB = _i2c->read();
+  }
 }
 
 void FuelGauge::writeRegister(byte address, byte MSB, byte LSB) {
-#if Wiring_Wire3
-	Wire3.beginTransmission(MAX17043_ADDRESS);
-    Wire3.write(address);
-    Wire3.write(MSB);
-    Wire3.write(LSB);
-    Wire3.endTransmission(true);
-#endif
-}
+  if (_i2c) {
```

Comment:
are the pointer checks necessary? static analysis shows that they are always initialized from a reference and can never be null.

Reply:
maybe not for `FuelGauge::begin()` but if `FuelGauge::begin()` wasn't called prior any function that uses the private `_i2c` pointer it'll fail.
I'll update my PR accordingly (or file a new one, since I'm not sure how to update ;-))